From e6d9556775b0b8f597ce48dc18c908b3cc09e2a3 Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Date: Wed, 21 Jul 2021 23:47:26 +0200
Subject: [PATCH] configure.ac: fix build without C++
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fix the following build failure raised on toolchains without C++:

  CXX      examples/find_links.o
/bin/bash: line 1: no: command not found
  CXX      examples/clean_text.o
/bin/bash: line 1: no: command not found
make[2]: *** [Makefile:1071: examples/find_links.o] Error 127
  CXX      examples/positions_of_class.o
make[2]: *** Waiting for unfinished jobs....
  CC       examples/get_title.o
/bin/bash: line 1: no: command not found
make[2]: *** [Makefile:1071: examples/clean_text.o] Error 127
make[2]: *** [Makefile:1071: examples/positions_of_class.o] Error 127
  CXX      examples/serialize.o
/bin/bash: line 1: no: command not found
  CXX      examples/prettyprint.o
make[2]: *** [Makefile:1071: examples/serialize.o] Error 127
/bin/bash: line 1: no: command not found
make[2]: *** [Makefile:1071: examples/prettyprint.o] Error 127

Fixes:
 - http://autobuild.buildroot.org/results/a32b5d3b959433fd5c3543661c37f80d27fbd010

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
[Raphaël: refresh patch on top of 0.13.1, update description]
Signed-off-by: Raphaël Mélotte <raphael.melotte@mind.be>
[Upstream status: https://codeberg.org/gumbo-parser/gumbo-parser/pulls/28]
---
 Makefile.am  | 2 ++
 configure.ac | 4 ++++
 2 files changed, 6 insertions(+)

diff --git a/Makefile.am b/Makefile.am
index e3f9ed7..8a93854 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -96,7 +96,9 @@ gumbo_test_DEPENDENCIES += gtest/lib/libgtest.a gtest/lib/libgtest_main.a
 gumbo_test_LDADD += gtest/lib/libgtest.a gtest/lib/libgtest_main.a
 endif
 
+if HAVE_EXAMPLES
 noinst_PROGRAMS = clean_text find_links get_title positions_of_class benchmark serialize prettyprint
+endif
 LDADD = libgumbo.la
 AM_CPPFLAGS = -I"$(srcdir)/src"
 
diff --git a/configure.ac b/configure.ac
index 5636a20..6d21b95 100644
--- a/configure.ac
+++ b/configure.ac
@@ -28,6 +28,10 @@ AC_CHECK_LIB([gtest_main],
              AM_CONDITIONAL(HAVE_SHARED_LIBGTEST, [true]),
              AM_CONDITIONAL(HAVE_SHARED_LIBGTEST, [false]))
 
+AC_ARG_ENABLE([examples],
+              AS_HELP_STRING([--disable-examples], [Disable examples]))
+AM_CONDITIONAL([HAVE_EXAMPLES], [test "x$enable_examples" != "xno"])
+
 # Init Automake & libtool
 AM_INIT_AUTOMAKE([foreign subdir-objects])
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
-- 
2.49.0

